### ISSUE TITLE
postgresql extension migrations fail

### ISSUE THREAD

**Issue by jfrazx** [rails/rails#27467]
### Steps to reproduce

Create a migration for a Postgresql extension (new or existing project)

then: `rake db:migrate`

I attempted this with extensions 'uuid-ossp' and 'postgis' with the same results. 

```ruby
class EnableUuidExtension < ActiveRecord::Migration[5.0]
  def change
    enable_extension 'uuid-ossp'
  end
end
```

### Expected behavior
The migration should complete successfully and enable the Postgresql extension.

```shell
== 20161227010639 EnableUuidExtension: migrating ==============================
-- enable_extension("uuid-ossp")
   -> 0.1001s
== 20161227010639 EnableUuidExtension: migrated (0.1002s) =====================
```

### Actual behavior
Migration fails.

`rake db:migrate `

```
== 20161227055512 EnableUuidExtension: migrating ==============================
-- enable_extension("uuid-ossp")
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

PG::SyntaxError: ERROR:  syntax error at or near "PARALLEL"
: CREATE EXTENSION IF NOT EXISTS "uuid-ossp"
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:598:in `async_exec'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:598:in `block in exec_no_cache'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract_adapter.rb:589:in `block in log'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activesupport-5.0.1/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract_adapter.rb:583:in `log'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:598:in `exec_no_cache'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:585:in `execute_and_clear'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:103:in `exec_query'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:333:in `enable_extension'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:846:in `block in method_missing'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:815:in `block in say_with_time'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:815:in `say_with_time'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:835:in `method_missing'
/Users/jfrazx/Documents/projects/rails5_test/db/migrate/20161227055512_enable_uuid_extension.rb:3:in `change'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:789:in `exec_migration'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:773:in `block (2 levels) in migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:772:in `block in migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:398:in `with_connection'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:771:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:951:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1212:in `block in execute_migration_in_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1280:in `block in ddl_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `block in transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract/transaction.rb:189:in `within_new_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/transactions.rb:211:in `transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1280:in `ddl_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1211:in `execute_migration_in_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1184:in `block in migrate_without_lock'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1183:in `each'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1183:in `migrate_without_lock'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1132:in `block in migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1299:in `with_advisory_lock'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1132:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1006:in `up'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:984:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/tasks/database_tasks.rb:161:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/railties/databases.rake:58:in `block (2 levels) in <top (required)>'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
ActiveRecord::StatementInvalid: PG::SyntaxError: ERROR:  syntax error at or near "PARALLEL"
: CREATE EXTENSION IF NOT EXISTS "uuid-ossp"
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:598:in `async_exec'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:598:in `block in exec_no_cache'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract_adapter.rb:589:in `block in log'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activesupport-5.0.1/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract_adapter.rb:583:in `log'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:598:in `exec_no_cache'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:585:in `execute_and_clear'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:103:in `exec_query'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:333:in `enable_extension'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:846:in `block in method_missing'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:815:in `block in say_with_time'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:815:in `say_with_time'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:835:in `method_missing'
/Users/jfrazx/Documents/projects/rails5_test/db/migrate/20161227055512_enable_uuid_extension.rb:3:in `change'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:789:in `exec_migration'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:773:in `block (2 levels) in migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:772:in `block in migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:398:in `with_connection'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:771:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:951:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1212:in `block in execute_migration_in_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1280:in `block in ddl_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `block in transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract/transaction.rb:189:in `within_new_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/transactions.rb:211:in `transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1280:in `ddl_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1211:in `execute_migration_in_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1184:in `block in migrate_without_lock'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1183:in `each'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1183:in `migrate_without_lock'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1132:in `block in migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1299:in `with_advisory_lock'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1132:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1006:in `up'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:984:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/tasks/database_tasks.rb:161:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/railties/databases.rake:58:in `block (2 levels) in <top (required)>'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
PG::SyntaxError: ERROR:  syntax error at or near "PARALLEL"
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:598:in `async_exec'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:598:in `block in exec_no_cache'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract_adapter.rb:589:in `block in log'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activesupport-5.0.1/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract_adapter.rb:583:in `log'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:598:in `exec_no_cache'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:585:in `execute_and_clear'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:103:in `exec_query'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/postgresql_adapter.rb:333:in `enable_extension'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:846:in `block in method_missing'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:815:in `block in say_with_time'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:815:in `say_with_time'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:835:in `method_missing'
/Users/jfrazx/Documents/projects/rails5_test/db/migrate/20161227055512_enable_uuid_extension.rb:3:in `change'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:789:in `exec_migration'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:773:in `block (2 levels) in migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:772:in `block in migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:398:in `with_connection'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:771:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:951:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1212:in `block in execute_migration_in_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1280:in `block in ddl_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `block in transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract/transaction.rb:189:in `within_new_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/transactions.rb:211:in `transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1280:in `ddl_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1211:in `execute_migration_in_transaction'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1184:in `block in migrate_without_lock'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1183:in `each'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1183:in `migrate_without_lock'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1132:in `block in migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1299:in `with_advisory_lock'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1132:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:1006:in `up'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/migration.rb:984:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/tasks/database_tasks.rb:161:in `migrate'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/activerecord-5.0.1/lib/active_record/railties/databases.rake:58:in `block (2 levels) in <top (required)>'
/Users/jfrazx/.rvm/gems/ruby-2.3.1@rails5test/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
```

### System configuration

Postgresql version:
`psql --version`
=> psql (PostgreSQL) 9.6.1

**Rails version**:
`rails --version`
=> Rails 5.0.1

**Ruby version**:
`ruby --version`
=> ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin15]

**Comment by matthewd** [comment]
> ```
> ERROR:  syntax error at or near "PARALLEL"
> : CREATE EXTENSION IF NOT EXISTS "uuid-ossp"
> ```

If you run `CREATE EXTENSION IF NOT EXISTS "uuid-ossp"` directly in `psql`, does it work? It sounds like you have a (probably version mismatch) issue in your PostgreSQL setup.

**Comment by markburns** [comment]
FYI for others with this issue

```
brew postgresql-upgrade-database
gem uninstall pg 
gem install pg
```

Fixed the issue for me with Rails 5.2.0, High Sierra, and going from postgres 9.x to 10.3
